src/charm.py

```diff
@@ -1280,6 +1280,17 @@ def _on_update_status(self, _) -> None:
             logger.debug("on_update_status early exit: Service has not been added nor started yet")
             return
 
+        if (
+            "restoring-backup" not in self.app_peer_data
+            and services[0].current != ServiceStatus.ACTIVE
+        ):
+            logger.warning(
+                "on_update_status early exit: pebble service inactive, restarting container"
+            )
+            self.unit.status = MaintenanceStatus("Database service inactive, restarting")
```

Comment:
Should we really change the status here? Assuming the `container.restart()` works out we should be back to active on the next line, while the new status will remain until another hook clears it up.

Reply:
Supposing we don't use `return` inside this if statement, I guess the container restart would work and resolve the situation in time before the `set_active_status` by the end of the hook, the problem is: What if it doesn't resolve it in time? what if the restart doesn't fix the underlying problem? I thought about using this status just to let the user know what's happening, and not keep the normal active status in such worse cases.

One thing we could do is just removing the `return` inside the if, though I'm not sure it would make sense to proceed with the hook execution right after a container restart.

pinging @marceloneppel too

Reply:
For this specific situation, @lucasgameiroborges, you may create a `tenacity.Retrying` block to check if the service started back up again (through the result from `self._patroni.member_started`). If it's okay, you can call `self._set_active_status()`. Otherwise, return and wait for the next hook that fixes the status.

Reply:
Implemented what you suggested @marceloneppel with the slight difference that, instead of explicitly calling `self._set_active_status()` I just let the hook finish and set the status in the final line, like normal. LMK if it makes sense.

Reply:
That's fine for me. Thanks!